meterpreter/meterpreter/src/main/java/com/metasploit/meterpreter/Meterpreter.java

```diff
@@ -41,6 +41,92 @@
     private List/* <byte[]> */tlvQueue = null;
 
 
+    private final TransportList transports = new TransportList();
+    private int ignoreBlocks = 0;
+    private byte[] uuid;
+    private long sessionExpiry;
+
+
+    private void loadConfiguration(DataInputStream in, OutputStream rawOut, byte[] configuration) throws MalformedURLException {
+        // socket handle is 4 bytes, followed by exit func, both of
+        // which we ignore.
+        int csr = 8;
+
+        // We start with the expiry, which is a 32 bit int
+        setExpiry(unpack32(configuration, csr));
+        csr += 4;
+
+        // this is followed with the UUID
+        this.uuid = readBytes(configuration, csr, UUID_LEN);
+        csr += UUID_LEN;
+
+        // here we need to loop through all the given transports, we know that we're
+        // going to get at least one.
+        while (configuration[csr] != '\0') {
+            // read the transport URL
+            String url = readString(configuration, csr, URL_LEN);
+            csr += URL_LEN;
+
+            Transport t = null;
+            if (url.startsWith("tcp")) {
+                t = new TcpTransport(url);
+            } else {
+                t = new HttpTransport(url);
+            }
+
+            csr = t.parseConfig(configuration, csr);
+            if (this.transports.isEmpty()) {
+                t.bind(in, rawOut);
+            }
+            this.transports.add(t);
+        }
+        
+        // we don't currently support extensions, so when we reach the end of the
+        // list of transports we just bomb out.
+    }
+
+    public byte[] getUUID() {
+        return this.uuid;
+    }
+
+    public long getExpiry() {
+        return (this.sessionExpiry - System.currentTimeMillis()) / Transport.MS;
+    }
+
+    public int getIgnoreBlocks() {
+        return this.ignoreBlocks;
+    }
+
+    public void setExpiry(long seconds) {
+        this.sessionExpiry = System.currentTimeMillis() + seconds * Transport.MS;
+    }
+
+    public void sleep(long milliseconds) {
+        try {
+            Thread.sleep(milliseconds);
+        } catch (InterruptedException ex) {
+            // ignore
+        }
+    }
+
+    public static String readString(byte[] bytes, int offset, int size) {
+        return new String(readBytes(bytes, offset, size)).trim();
```

Comment:
So it just depends on what you want to happen in case it is not ascii. If you want question marks, use "US-ASCII", if you want UTF-8 use "UTF-8". "ISO-8859-1" would be another option, and has the advantage that it is the fastest (not that it would matter in that context) and it is reversible if needed.

I personally would choose "UTF-8" (both here and on the Framework side).